nodebuilder/tests/tastora/blob_test.go

```diff
@@ -31,319 +34,432 @@ func TestBlobTestSuite(t *testing.T) {
+
+	// Wait for both blocks
+	maxHeight := height1
+	if height2 > maxHeight {
+		maxHeight = height2
+	}
+
+	_, err = bridgeClient.Header.WaitForHeight(ctx, maxHeight)
 	s.Require().NoError(err)
-	s.Require().Len(retrievedBlobs, 2)
 
-	// Verify individual blob retrieval
-	for i, nodeBlob := range nodeBlobs {
-		retrievedBlob, err := client.Blob.Get(ctx, height, namespace, nodeBlob.Commitment)
-		s.Require().NoError(err)
-		s.Require().NotNil(retrievedBlob)
+	// Allow propagation time
+	time.Sleep(10 * time.Second)
 
-		retrievedData := bytes.TrimRight(retrievedBlob.Data(), "\x00")
-		if i == 0 {
-			s.Assert().Equal(data1, retrievedData)
-		} else {
-			s.Assert().Equal(data2, retrievedData)
-		}
+	clients := map[string]*client.Client{
+		"bridge": bridgeClient,
+		"light":  lightClient,
+	}
+
+	// Test namespace isolation on both nodes
+	for nodeType, client := range clients {
+		s.Run(fmt.Sprintf("NamespaceIsolation_%s", nodeType), func() {
+			// Ensure node is synced
+			_, err := client.Header.WaitForHeight(ctx, maxHeight)
+			s.Require().NoError(err)
+
+			// Get all blobs for namespace1 - should only return namespace1 data
+			allBlobs1, err := client.Blob.GetAll(ctx, height1, []share.Namespace{namespace1})
+			s.Require().NoError(err, "%s should get namespace1 blobs", nodeType)
+			s.Assert().Len(allBlobs1, 1, "%s should find exactly 1 blob in namespace1", nodeType)
+
+			if len(allBlobs1) > 0 {
+				retrievedData1 := bytes.TrimRight(allBlobs1[0].Data(), "\x00")
+				s.Assert().Equal(data1, retrievedData1, "%s namespace1 data should match", nodeType)
+			}
+
+			// Get all blobs for namespace2 - should only return namespace2 data
+			allBlobs2, err := client.Blob.GetAll(ctx, height2, []share.Namespace{namespace2})
+			s.Require().NoError(err, "%s should get namespace2 blobs", nodeType)
+			s.Assert().Len(allBlobs2, 1, "%s should find exactly 1 blob in namespace2", nodeType)
+
+			if len(allBlobs2) > 0 {
+				retrievedData2 := bytes.TrimRight(allBlobs2[0].Data(), "\x00")
+				s.Assert().Equal(data2, retrievedData2, "%s namespace2 data should match", nodeType)
+			}
+
+			// Cross-namespace verification - namespace1 query should not return namespace2 data
+			crossBlobs, err := client.Blob.GetAll(ctx, height2, []share.Namespace{namespace1})
+			s.Require().NoError(err)
+			// Should be empty since namespace1 blob was submitted at height1, not height2
+			s.Assert().Empty(crossBlobs, "%s should not find namespace1 blobs at height2", nodeType)
+
+			s.T().Logf("✅ %s namespace isolation verified", nodeType)
+		})
 	}
 }
 
-// TestBlobGet_ExistingBlob tests blob retrieval API for existing blobs
-func (s *BlobTestSuite) TestBlobGet_ExistingBlob() {
-	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+// TestConcurrentBlobOperations validates that multiple concurrent blob operations
+// work correctly across different nodes without conflicts
+func (s *BlobTestSuite) TestConcurrentBlobOperations() {
```

Comment:
The value in this test lies in the logic that tests whether BN can submit blobs concurrently. 

let's limit this test to that + also include a test case for LN to be able to do the same (not like they're any different from a state operations perspective, but better to test both just to be able to catch regressions)